http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -894,16 +896,21 @@ private <I, O, E> Mono<HttpResponse<O>> exchange(io.micronaut.http.HttpRequest<I
         AtomicReference<ScheduledExecutorService> scheduler = new AtomicReference<>(connectionManager.getGroup());
         ExecutionFlow<HttpResponse<O>> mono = resolveRequestURI(request).flatMap(uri -> {
             MutableHttpRequest<?> mutableRequest = toMutableRequest(request).uri(uri);
+            Object body = mutableRequest.getBody().orElse(null);
+            if (body instanceof Publisher<?> publisher) {
+                return ReactiveExecutionFlow.fromPublisher(publisher).flatMap(resolvedBody -> {
```

Comment:
I think Publisher support also needs multi-element support.

Reply:
Not sure how we should support it

Reply:
My vote is that we don't for now. Actually mirroring the server behavior is more complicated than this. But this implementation is not workable, e.g. `@Body Publisher<byte[]>` will support streaming on the server but not on the client.